pkgs/development/haskell-modules/with-packages-wrapper.nix

```diff
@@ -106,7 +106,7 @@ symlinkJoin {
 
   '' + (lib.optionalString (stdenv.targetPlatform.isDarwin && !isGhcjs && !stdenv.targetPlatform.isiOS) ''
     # Work around a linker limit in macOS Sierra (see generic-builder.nix):
-    local packageConfDir="$out/lib/${ghc.name}/package.conf.d";
+    local packageConfDir="${libDir}/package.conf.d";
```

Comment:
What are these changes for?

Reply:
Haha. I've done a `s#$out/lib/${ghc.name}#${libDir}#` and obviously it found another instance.

I think it is safe, because it looks coherent, but becuase it is inside a special case for darwin, I think a review by someone using ghc on darwin will be appreciated.